federation/pkg/dnsprovider/providers/aws/route53/zones.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package route53
+
+import (
+	"github.com/aws/aws-sdk-go/service/route53"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+)
+
+type Zones struct {
+	interface_ *Interface
+}
+
+func (zones Zones) List() ([]dnsprovider.Zone, error) {
+	input := route53.ListHostedZonesInput{}
+	response, err := zones.interface_.service.ListHostedZones(&input)
+	if err != nil {
+		return []dnsprovider.Zone{}, err
+	}
+	hostedZones := response.HostedZones // TODO: Handle result truncation
```

Comment:
AWS route53 can respond with paginated results.  By default, if there are more than 100 hosted zones, it returns them in pages.  Right now we only handle the first page, which is bad, but easily fixable, and we don't generally expect there to be more than one, or a small handful.  Make sense.